libs/angular-routing/src/lib/router.service.ts

```diff
@@ -47,6 +47,10 @@ export class Router {
   }
 
   serializeUrl(url: string, queryParams?: Params, hash?: string) {
+    // if relative path
+    if (!url.startsWith('/')) {
+      url = this.urlParser.joinUrls(this.location.path(), url);
```

Comment:
The URL Web API supports building relative URLs, so we can leverage that. WDYT?

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

```suggestion
      url = this.urlParser.parse(url, this.getLocation()).pathname;
```

Reply:
I've looked into that solution first. There is one fine difference to the current implementation:
- The `joinUrls` keeps query params and hash from the target URL making functions like `serializeUrl` (and `go`) work without the need of explicit path and params separation

Let me know what you think.

Reply:
I tested it with passing query params and a hash, and it parsed it correctly. That being said, it provides a little less overhead if you want to use it separately.